src/sidebar/components/menu.js

```diff
@@ -82,32 +82,72 @@ function Menu({
       return () => {};
     }
 
-    const removeListeners = listen(
+    // Close menu when user presses Escape key, regardless of focus.
+    const removeKeypressListener = listen(
       document.body,
-      ['keypress', 'click', 'mousedown'],
+      ['keypress'],
       event => {
-        if (event.type === 'keypress' && event.key !== 'Escape') {
-          return;
+        if (event.key === 'Escape') {
+          closeMenu();
         }
-        if (
-          event.type === 'mousedown' &&
-          menuRef.current &&
-          menuRef.current.contains(event.target)
-        ) {
-          // Close the menu as soon as the user _presses_ the mouse outside the
-          // menu, but only when they _release_ the mouse if they click inside
-          // the menu.
-          return;
-        }
-        closeMenu();
       }
     );
 
-    return removeListeners;
+    // Close menu if user focuses an element outside the menu via any means
+    // (key press, programmatic focus change).
+    const removeFocusListener = listen(
+      document.body,
+      'focus',
+      event => {
+        if (!menuRef.current.contains(event.target)) {
+          closeMenu();
+        }
+      },
+      { useCapture: true }
```

Comment:
Can you walk me through why it's necessary to use `useCapture` here?

Reply:
There are two reasons. The first is that the `focus` event does not bubble up from the target element (although [focusin](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusin_event) does, so it would be an alternative. I thought I checked last week and saw there was a browser support issue but I checked again just now [and it looks like all of our target browsers do support it](https://caniuse.com/#feat=focusin-focusout-events)). The other is that it didn't seem to me like the receiving element should be able to cancel this effect (eg. via `stopPropagation`).